qa/tasks/cephfs/test_misc.py

```diff
@@ -276,23 +276,23 @@ def _test_sync_stuck_for_around_5s(self, dir_path, file_sync=False):
                 self.mount_a.run_shell(["mkdir", os.path.join(dir_path, f"{i}_{j}")])
             start = time.time()
             if file_sync:
-                self.mount_a.run_shell(['python3', '-c', sync_dir_pyscript])
+                self.mount_a.run_shell(['python3', '-c', sync_dir_pyscript], timeout=4)
             else:
-                self.mount_a.run_shell(["sync"])
+                self.mount_a.run_shell(["sync"], timeout=4)
+            # the real duration should be less than the rough one
             duration = time.time() - start
-            log.info(f"sync mkdir i = {i}, duration = {duration}")
-            self.assertLess(duration, 4)
+            log.info(f"sync mkdir i = {i}, rough duration = {duration}")
 
             for j in range(5):
                 self.mount_a.run_shell(["rm", "-rf", os.path.join(dir_path, f"{i}_{j}")])
             start = time.time()
             if file_sync:
-                self.mount_a.run_shell(['python3', '-c', sync_dir_pyscript])
+                self.mount_a.run_shell(['python3', '-c', sync_dir_pyscript], timeout=4)
             else:
-                self.mount_a.run_shell(["sync"])
+                self.mount_a.run_shell(["sync"], timeout=4)
```

Comment:
since the assertion failed because it took 6.77 seconds, shouldn't the timeout be kept a bit lenient? something around 10 secs?

Reply:
NO. The `sync` or `fsync`, which will wait the `mdlog` flushing in MDS to finish, should be very fast normally, else the bug we fixed comes back.

Why it took `6.77` is because the `sync` command itself was scheduled late, not the `sync` command itself took that long. From the client logs it finished very fast in 0.1s.